kernel/src/upcall.rs

```diff
@@ -78,21 +79,46 @@ impl Upcall {
         r0: usize,
         r1: usize,
         r2: usize,
-    ) -> bool {
-        let res = self.fn_ptr.map_or(false, |fp| {
-            process.enqueue_task(process::Task::FunctionCall(process::FunctionCall {
-                source: process::FunctionCallSource::Driver(self.upcall_id),
-                argument0: r0,
-                argument1: r1,
-                argument2: r2,
-                argument3: self.appdata,
-                pc: fp.as_ptr() as usize,
-            }))
-        });
+    ) -> Result<(), UpcallError> {
+        let res = self.fn_ptr.map_or(
+            Ok(()), // A null-Upcall is treated as being delivered to
+            // the process and ignored
```

Comment:
```suggestion
            // A null-Upcall is treated as being delivered to
            // the process and ignored
            Ok(()),
```

Reply:
Ah! That was so well aligned before `rustfmt` came along. :)